geoengine/datasets.py

```diff
@@ -404,3 +540,60 @@ class StoredDataset(NamedTuple):
 
     dataset_id: DatasetId
     upload_id: UploadId
+
+
+@dataclass
+class Volume:
+    '''A volume'''
+
+    id: VolumeId
+    path: str
+
+    @classmethod
+    def from_response(cls, response: Dict[str, str]) -> Volume:
+        '''Parse a http response to an `Volume`'''
+        print(response)
+
+        return Volume(VolumeId(UUID(response['id'])), response['path'])
+
+
+def volumes(timeout: int = 60) -> List[Volume]:
+    '''Returns a list of all volumes'''
+
+    session = get_session()
+
+    response = req.get(f'{session.server_url}/dataset/volumes',
+                       headers=session.admin_auth_header,
+                       timeout=timeout
+                       ).json()
+
+    return [Volume.from_response(v) for v in response]
```

Comment:
I found a nice way for Jupyter was to implement `_repr_html_()`.

Then, the volume listing looks good, e.g. as a table.

Reply:
this method returns a List. To implement `_repr_html_()` I guess I would have to implement a custom type like `VolumeList` right? I don't know if we really need that but I can implement it if you want.

Reply:
I would leave that up to you. I just found that it gives our library a "pro look" when used in Jupyter.

Reply:
I don't think we need it for the volumes.